### PR TITLE
Fix a subtle issue with PropertyChanged in ViewModels

### PR DESCRIPTION
**Pull request by zadjii-msft**
While debugging #13694, we discovered a very subtle bug we had accidentally introduced in a few places. `ViewModelHelper` defines a `PropertyChanged` event, backed by a `_propertyChangedHandlers` `event`. All opbservable properties in the viewmodels are supposed to run through that event. However, if you do `WINRT_CALLBACK(PropertyChanged, Windows::UI::Xaml::Data::PropertyChangedEventHandler)` in a derived class, it'll override that original method with the new one. XAML will subscribe to the second one, which is backed by `_PropertyChangedHandlers`, but the properties will still raise notifications on the callbacks registered to `_propertyChangedHandlers`. 

This change makes it more explicit in these derived classes, that the `PropertyChanged` method exposed by these classes is indeed the one that's implemented in the base class. 

This is a bit of a footgun, for sure. AuditMode would have apparently caught this, as we'd be overriding a method without using the `override` keyword.

Unblocks #13694.

### CODE REVIEW

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
@carlos-zamora  you will need to propagate some of these changes to the new ColorScheme page!

**Comment by DHowett** [comment]
I converted this to a draft so that it would not merge without a description

File: src/cascadia/TerminalSettingsEditor/ColorSchemeViewModel.h
Comment (lhecker) [inline src/cascadia/TerminalSettingsEditor/ColorSchemeViewModel.h:31]: (This should be "why" and not "what".)

**Review by lhecker — APPROVED** [review]

**Review by carlos-zamora — APPROVED** [review]
Once this merges, I'll merge it into #13269 and update that PR accordingly

**Pull request merged**
